x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/type/Types.java

```diff
@@ -48,8 +48,12 @@
 
     private static DataType getType(DataTypeRegistry typeRegistry, Map<String, Object> content) {
         if (content.containsKey("type")) {
+            String typeName = content.get("type").toString();
+            if ("wildcard".equals(typeName)) {
```

Comment:
Why is this needed? Aren't wildcard returned as keywords? Or is this using a different API/source?

Reply:
That's for testing. We use .json files having an index' complete mapping (see `org.elasticsearch.xpack.ql.type.TypesTests`) and the `wildcard` comparison is for bypassing SQL's and EQL's specific data types auto-resolver in the specific case of `wildcard`.